### PR TITLE
Implement scheduled bakes

### LINKED ISSUE #4: Scheduled bakes
For each recipe, you should have the option of specifying "bake this every N days"

### PR DESCRIPTION
**Pull request by cb372**
* Add a scheduler based on Quartz
* Add the corresponding form field to the UI

I decided to just let users input a raw cron expression in the UI. We're all adults here.

Fixes #4

### CODE REVIEW

File: app/components/AppComponents.scala [hunk 1]
```diff
@@ -78,10 +79,19 @@ class AppComponents(context: Context)
 
   val prism = new Prism(wsClient)
 
+  val scheduledBakeRunner = {
+    val enabled = identity.stage == "PROD" // don't run scheduled bakes on dev machines
```
Comment (adamnfish) [inline app/components/AppComponents.scala]: Is the compile-time-DI wiring class the place for logic like this?

File: app/schedule/BakeScheduler.scala [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import org.quartz.impl.StdSchedulerFactory
+import org.quartz.{ TriggerKey, JobKey, JobDataMap }
+import play.api.Logger
+
+class BakeScheduler(scheduledBakeRunner: ScheduledBakeRunner) {
+
+  private val scheduler = StdSchedulerFactory.getDefaultScheduler()
+
+  def initialise(recipes: Iterable[Recipe]): Unit = {
+    recipes.flatMap(r => r.bakeSchedule.map(s => (r.id, s))).foreach {
```
Comment (adamnfish) [inline app/schedule/BakeScheduler.scala]: It seems like one of the `map` and `foreach` is redundant here - all the map is doing is putting the data that's already in scope into a tuple that you then immediately extract. Would this be clearer as just a nested foreach?

```scala
recipes.foreach { recipe =>
  recipe.bakeSchedule.foreach { schedule =>
    scheduleBake(recipe.id, schedule)
  }
}
```

**Comment by philwills** [comment]
Dammit, I'd just got the Scanamo upgrade PR rebased against master. 😿

**Comment by cb372** [comment]
You snooze, you lose!

**Pull request merged**